README.rst
:**NOTE**: **This github repository is archived. The repository contents and history have moved to** `google-cloud-python`_.

.. _google-cloud-python: https://github.com/googleapis/google-cloud-python/tree/main/packages/google-cloud-bigquery-logging


Python Client for BigQuery Logging Protos API
=============================================

|stable| |pypi| |versions|

`BigQuery Logging Protos API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-bigquery-logging.svg
   :target: https://pypi.org/project/google-cloud-bigquery-logging/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-bigquery-logging.svg
   :target: https://pypi.org/project/google-cloud-bigquery-logging/
.. _BigQuery Logging Protos API: https://cloud.google.com/bigquery/docs/reference/auditlogs
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/bigquerylogging/latest
.. _Product Documentation:  https://cloud.google.com/bigquery/docs/reference/auditlogs

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the BigQuery Logging Protos API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the BigQuery Logging Protos API.:  https://cloud.google.com/bigquery/docs/reference/auditlogs
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current `active`_ and `maintenance`_ versions of
Python.

Python >= 3.7

.. _active: https://devguide.python.org/devcycle/#in-development-main-branch
.. _maintenance: https://devguide.python.org/devcycle/#maintenance-branches

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an `end-of-life`_
version of Python, we recommend that you update as soon as possible to an actively supported version.

.. _end-of-life: https://devguide.python.org/devcycle/#end-of-life-branches

Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-bigquery-logging


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-bigquery-logging

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for BigQuery Logging Protos API
   to see other available methods on the client.
-  Read the `BigQuery Logging Protos API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _BigQuery Logging Protos API Product documentation:  https://cloud.google.com/bigquery/docs/reference/auditlogs
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

### [1.0.1](https://www.github.com/googleapis/python-bigquery-logging/compare/v1.0.0...v1.0.1) (2021-11-01)

### Bug Fixes

* fix(deps): drop packaging dependency ([a5b1978](https://www.github.com/googleapis/python-bigquery-logging/commit/a5b19782e2ac30f5de150fb873d777fa121bd0ad))
* fix(deps): require google-api-core >= 1.28.0 ([a5b1978](https://www.github.com/googleapis/python-bigquery-logging/commit/a5b19782e2ac30f5de150fb873d777fa121bd0ad))


### Documentation

* list oneofs in docstring ([a5b1978](https://www.github.com/googleapis/python-bigquery-logging/commit/a5b19782e2ac30f5de150fb873d777fa121bd0ad))

## [1.0.0](https://www.github.com/googleapis/python-bigquery-logging/compare/v0.3.0...v1.0.0) (2021-10-12)


### Features

* bump release level to production/stable ([#41](https://www.github.com/googleapis/python-bigquery-logging/issues/41)) ([19db3eb](https://www.github.com/googleapis/python-bigquery-logging/commit/19db3eb189579fb1c71ff94b3db4ffd279792fe7))

## [0.3.0](https://www.github.com/googleapis/python-bigquery-logging/compare/v0.2.2...v0.3.0) (2021-10-08)


### Features

* add context manager support in client ([#43](https://www.github.com/googleapis/python-bigquery-logging/issues/43)) ([493010c](https://www.github.com/googleapis/python-bigquery-logging/commit/493010cbfb288a75a58761d5281993009013e1b6))

### [0.2.2](https://www.github.com/googleapis/python-bigquery-logging/compare/v0.2.1...v0.2.2) (2021-09-24)


### Bug Fixes

* add 'dict' annotation type to 'request' ([546c1b3](https://www.github.com/googleapis/python-bigquery-logging/commit/546c1b3539f03a172eed2cdf202615a5fa37418f))

### [0.2.1](https://www.github.com/googleapis/python-bigquery-logging/compare/v0.2.0...v0.2.1) (2021-07-29)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#22](https://www.github.com/googleapis/python-bigquery-logging/issues/22)) ([c18bc2d](https://www.github.com/googleapis/python-bigquery-logging/commit/c18bc2da92e0823178e59804d4ac2c2824feb3a7))


### Documentation

* add Samples section to CONTRIBUTING.rst ([#18](https://www.github.com/googleapis/python-bigquery-logging/issues/18)) ([c157b7f](https://www.github.com/googleapis/python-bigquery-logging/commit/c157b7febc2e2e62bc6dbd8d6b02ef3aa6ac2c3f))


### Miscellaneous Chores

* release as 0.2.1 ([#23](https://www.github.com/googleapis/python-bigquery-logging/issues/23)) ([977f2ab](https://www.github.com/googleapis/python-bigquery-logging/commit/977f2abeb7c52729ba6159f77b2a30a3015ed82c))

## [0.2.0](https://www.github.com/googleapis/python-bigquery-logging/compare/v0.1.1...v0.2.0) (2021-07-14)


### Features

* add always_use_jwt_access ([#12](https://www.github.com/googleapis/python-bigquery-logging/issues/12)) ([17140f0](https://www.github.com/googleapis/python-bigquery-logging/commit/17140f0468a66948c07c9583a031598b50a9bc03))


### Documentation

* omit mention of Python 2.7 in 'CONTRIBUTING.rst' ([#1127](https://www.github.com/googleapis/python-bigquery-logging/issues/1127)) ([#7](https://www.github.com/googleapis/python-bigquery-logging/issues/7)) ([53f6808](https://www.github.com/googleapis/python-bigquery-logging/commit/53f6808fa2b317ee5f9a4ca0a54214267cc1dce0)), closes [#1126](https://www.github.com/googleapis/python-bigquery-logging/issues/1126)

### [0.1.1](https://www.github.com/googleapis/python-bigquery-logging/compare/v0.1.0...v0.1.1) (2021-06-16)


### Bug Fixes

* exclude docs and tests from package ([#4](https://www.github.com/googleapis/python-bigquery-logging/issues/4)) ([f241715](https://www.github.com/googleapis/python-bigquery-logging/commit/f24171552220f5a120535c101e2ab61b62d752b5))

## 0.1.0 (2021-06-02)


### Features

* generate protos for google.cloud.biguqyery.logging.v1 ([3578ce8](https://www.github.com/googleapis/python-bigquery-logging/commit/3578ce8ce9889e93113efa5f004a18d894446e26))


owl-bot-staging/v1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")


nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
]

@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/bigquery_logging_v1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python='3.7')
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=['3.6', '3.7'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python='3.6')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
